crates/uv-auth/src/middleware.rs

```diff
@@ -48,6 +49,54 @@ impl NetrcMode {
     }
 }
 
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq)]
+pub enum AuthPolicy {
+    /// Attempt to authenticate if needed.
```

Comment:
We should probably be more detailed here. Like, we try an unauthenticated request and if that fails we'll try an authenticated request?